src/Router.sol

```diff
@@ -41,13 +41,16 @@ contract Router is IRouter, EIP712, FeeGenerator {
     /// @notice Address for invoking pause
     address public pauser;
 
+    /// @dev Modifier for blocking reentrancy by checking `currentUser`
+    ///      If reentrancy is not blocked, malicious contracts could transfer funds from agents and users
```

Comment:
Can you think of any attack vectors for transferring funds from the agent and user when the reentrancy check is not in place?

Reply:
I was thinking If the block is not placed, the flow could be router -> agent -> malicious contract -> router -> agent (transfer funds). But it only works when Router::execute get `user` from function inputs. If Router::execute takes `msg.sender` as `user`, the issue could be avoided. Suggest removing `If reentrancy ...`

Reply:
Agree. I think this reentrancy guard is in place to ensure that within the same transaction, the `user` does not get changed leading to the first router::execute referring to the incorrect `user`. I also haven't come up with a concrete way to exploit user's funds.

Reply:
fixed.